lib/kino.ex

```diff
@@ -193,4 +212,60 @@ defmodule Kino do
   def nothing() do
     :"do not show this result in output"
   end
+
+  @doc """
+  Starts a process under the Kino supervisor.
+
+  The process is automatically terminated when the current process
+  terminates or the current cell reevaluates.
+  """
+  @spec start_child(
+          Supervisor.child_spec()
+          | {module(), term()}
+          | module()
+        ) :: DynamicSupervisor.on_start_child()
+  def start_child(child_spec) do
+    # Starting a process that calls Kino.start_child/1 in its init
+    # would block forever, so we don't allow nesting
+    if Kino.DynamicSupervisor in Process.get(:"$ancestors", []) do
```

Comment:
Let's add a test for this to make sure it is triggered. Something like this should trigger it:

    Kino.start_child({Task, fn -> Kino.start_child({Task, fn -> :ok end}) end})

Reply:
Good call! We need the assertion in the inner call, so I made the test wait for the outer process to terminate, let me know if there's any better way :)